patient_data_source.py

```diff
@@ -0,0 +1,228 @@
+		YYYY-MM-DDThh:mm:ss+zz:zz as described https://build.fhir.org/datatypes.html#dateTime. Implementation is not required."""
+		return None
+
+class PatientDataSource(ABC):
+	"""Abstract class for loading patient data into a smart FHIR server"""
+	
+	@abstractmethod
+	def get_all_patients(self) -> Iterable[Patient]:
+		"""Returns a list of all Patients."""
+		pass
+
+	@abstractmethod
+	def get_patient_observations(self, patient : Patient) -> Iterable[Observation]:
+		"""Returns a list of all Observation for one patient."""
+		pass
+
+	def create_patient(self, patient: Patient) -> FHIR_Patient :
+		"""Create a smart FHIR_Patient object."""
+		gender = patient.get_gender().name.lower()
+		family, given = patient.get_name()
+		date = patient.get_dob()
+
+		fhir_patient_args = {
+		  'gender' : gender,
+		  'name' : [{'family':family,'given':[given]}],
+		}
+
+		if (date is not None):
+			fhir_patient_args['birthDate'] = date
+
+		identifier_system = patient.get_identifier_system()
+		identifier_value = patient.get_identifier_value()
+
+		if (identifier_system is not None and identifier_value is not None):
+			fhir_patient_args['identifier'] = [{
+				'system': identifier_system,
+				'value': identifier_value
+			}]
+		elif (identifier_system is not None):
+			fhir_patient_args['identifier'] = [{
+				'system': identifier_system
+			}]
+		elif (identifier_value is not None):
+			fhir_patient_args['identifier'] = [{
+				'value': identifier_value
+			}]
+
+		return FHIR_Patient(fhir_patient_args)
+
+	def create_observation(self, observation : Observation, patient_id : str) -> FHIR_Observation : 
+		"""Create a smart FHIR_Observation object. patient_id is an ID value of a Patient item currently on the FHIR server"""
+		value = observation.get_value()
+		unit_string = observation.get_unit_string()
+		unit_code = observation.get_unit_code()
+		code_value = observation.get_observation_code_value()
+		code_system = observation.get_observation_code_system()
+		display_string = observation.get_display_string()
+		date = observation.get_time()
+
+		fhir_observation_args = {
+		  'code' : {
+		    'coding' : [
+		      {'code': code_value, 'display': display_string, 'system': code_system}
+		    ]
+		  },
+		  'status' :'final',
+		  'subject': {'reference': f'Patient/{patient_id}'},
+		  'valueQuantity': {
+		    'code': unit_code,
+		    'system': 'http://unitsofmeasure.org',
+		    'unit': unit_string,
+		    'value': value
+		  },
+		}
+
+		if (date is not None):
+			fhir_observation_args['effectiveDateTime'] = date
+
+		identifier_system = observation.get_identifier_system()
+		identifier_value = observation.get_identifier_value()
+
+		if (identifier_system is not None and identifier_value is not None):
+			fhir_observation_args['identifier'] = [{
+				'system': identifier_system,
+				'value': identifier_value
+			}]
+		elif (identifier_system is not None):
```

Comment:
again, I think if there's a system but no value then there's no sense in putting anything into the observation dict